simulation_bridge/docs/inmemory_adapter.md

```diff
@@ -0,0 +1,100 @@
+# In-Memory Adapter
+
+The In-Memory Adapter enables the Simulation Bridge to operate entirely within a Python process without external message brokers. It provides direct communication between client code and simulation components through callback functions.
+
+## Architecture
+
+The adapter bypasses network protocols (RabbitMQ, MQTT, REST) by implementing signal-based communication using Python's Blinker library. All message routing occurs in-process, eliminating broker dependencies while maintaining the same interface as networked adapters.
+
+## API Reference
+
+### InMemorySimulation
+
+```python
+class InMemorySimulation:
+    def __init__(self, config_path: str | None = None)
+    def send(self, message: dict, callback: Callable[[dict], None]) -> None
+    def stop() -> None
+```
+
+#### Constructor
+
+- `config_path`: YAML configuration file path. If provided, configures bridge protocols and parameters.
+
+#### Methods
+
+- `send(message, callback)`: Submits simulation request. The callback receives result dictionaries as they are published.
+- `stop()`: Terminates adapters and disconnects signal handlers. Required to prevent resource leaks.
+
+## Message Format
+
+Requests must follow the standard simulation schema:
+
+```python
+{
+    "simulation": {
+        "request_id": str,     # Unique identifier (required)
+        "client_id": str,      # Client identifier
+        "simulator": str,      # Target simulator ("matlab", etc.)
+        "type": str,          # Execution type ("batch", "streaming")
+        "file": str,          # Simulation file path
+        "timestamp": str | None, # Optional ISO 8601 timestamp
+        "timeout": int | None,   # Optional max processing time (seconds)
+        "inputs": dict,       # Input parameters
+        "outputs": dict       # Expected output structure
+    }
+}
+```
+
+> Refer to the [Simulation Bridge User Guide](USERGUIDE.md) for detailed message structure and requirements.
+
+## Usage Example
+
+```python
+from simulation_bridge import InMemorySimulation  # Import the in-memory adapter
+import time  # Import time module for sleep functionality
+
+completed = False  # Global flag to track simulation completion
+
+def handle_result(msg):  # Callback function to process simulation results
+    global completed  # Access global completion flag
+    print("\nReceived:", msg)  # Print received message
+    if msg.get("status") == "completed":  # Check if simulation is complete
+        completed = True  # Set completion flag
+
+sim = InMemorySimulation("config.yaml")  # Create simulation instance with config
+
+sim.send({  # Send simulation request with message payload
+    "simulation": {
```

Comment:
The other protocol adapters receive `simulation.yaml` but this one is receiving a json. Please pass the yaml here as well.